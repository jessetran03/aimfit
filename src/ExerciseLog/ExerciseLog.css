.log-header {
  margin-top: 20px;
}

.add-entry-form {
  margin: 25px;
}

.add-entry-form input {
  margin-left: 7px;
  border-radius: 5px;
  padding: 3px;
  font-size: 15px;
  font-weight: 600;
}

.add-entry-form label {
  font-weight: 600;
}

.entry-rep-count-input, .entry-set-count-input {
  width: 35px;
  margin-bottom: 5px;
}

.entry-rep-count-input {
  margin-right: 21px;
}

.entry-weight-count-input {
  width: 102px;
  margin-bottom: 10px;
}