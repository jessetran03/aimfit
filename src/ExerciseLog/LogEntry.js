import React, { Component } from 'react'
import config from '../config'
import PropTypes from 'prop-types'
import moment from 'moment';

export default class LogEntry extends Component {
  static defaultProps = {
    getLogEntries: () => { },
  }
  static propTypes = {
    getLogEntries: PropTypes.func,
    id: PropTypes.number,
    set_count: PropTypes.number,
    rep_count: PropTypes.number,
    weight_count: PropTypes.number,
  }

  handleClickDelete = e => {
    e.preventDefault()
    const logEntryId = this.props.id

    fetch(`${config.API_ENDPOINT}/exercise_log/${logEntryId}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json'
      },
    })
      .then(() => {
        this.props.getExerciseLog()
      })
      .catch(error => {
        console.error({ error })
      })
  }

  render() {
    const { set_count, rep_count, weight_count, date_logged } = this.props
    return (
      <section className='workout-exercise'>
        <h3>
          {moment(date_logged).format('MM-DD-YYYY')}
        </h3>
        <p>
          {weight_count} lbs for {set_count}x{rep_count}
        </p>
        <button
          type='button'
          onClick={this.handleClickDelete}
        >
          Delete
        </button>
      </section>
    )
  }
}
