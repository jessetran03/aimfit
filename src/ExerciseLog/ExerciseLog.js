import React, { Component } from 'react'
import config from '../config'
import LogEntry from './LogEntry'
import './ExerciseLog.css'
import TokenService from '../services/token-service'

export default class ExerciseLog extends Component {
  state = {
    exerciseLog: [],
    exercise: {},
  }
  static defaultProps = {
    match: { params: {} },
    location: { pathname: {} }
  }

  componentDidMount() {
    this.getExercise()
    this.getExerciseLog()
  }

  getExerciseLog = () => {
    const { exercise_id } = this.props.match.params
    return fetch(`${config.API_ENDPOINT}/exercise_log/${exercise_id}`, {
      method: 'GET',
      headers: {
        'Authorization': `Bearer ${TokenService.getAuthToken()}`
      }
    })
      .then(res =>
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      )
      .then((exerciseLog) => {
        this.setState({ exerciseLog })
      })
      .catch(error => {
        console.error({ error })
      })
  }

  getExercise = () => {
    const { exercise_id } = this.props.match.params
    return fetch(`${config.API_ENDPOINT}/exercises/${exercise_id}`, {
      method: 'GET',
      headers: {
        'Authorization': `Bearer ${TokenService.getAuthToken()}`
      }
    })
      .then(res =>
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      )
      .then(exercise => {
        this.setState({ exercise })
      })
      .catch(error => {
        console.error({ error })
      })
  }

  handleAddEntry = e => {
    e.preventDefault()
    const { exercise_id } = this.props.match.params
    const newEntry = {
      set_count: e.target['entry-set-count'].value,
      rep_count: e.target['entry-rep-count'].value,
      weight_count: e.target['entry-weight-count'].value,
    }
    fetch(`${config.API_ENDPOINT}/exercise_log/${exercise_id}`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json',
        'authorization': `bearer ${TokenService.getAuthToken()}`
      },
      body: JSON.stringify(newEntry),
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(e => Promise.reject(e))
        return res.json()
      })
      .then(entry => {
        this.getExerciseLog()
        e.target['entry-set-count'].value = ''
        e.target['entry-rep-count'].value = ''
        e.target['entry-weight-count'].value = ''
      })
      .catch(error => {
        console.error({ error })
      })
  };

  handleClickBack = () => {
    this.props.history.goBack()
  };

  render() {
    const exerciseLog = this.state.exerciseLog
    const exercise = this.state.exercise
    return (
      <>
        <section className='workouts'>
          <button onClick={this.handleClickBack}>
            Back
          </button>
          <h2 className='log-header'>{exercise.exercise_name}</h2>
          <form onSubmit={this.handleAddEntry} className='add-entry-form'>
            <label htmlFor='entry-set-count-input' className='entry-set-count-label'>
              Set x Rep:
            </label>
            <input
              className='entry-set-count-input'
              type='number'
              min='0'
              id='entry-set-count-input'
              name='entry-set-count'
              required />
            &nbsp;x
            <input
              className='entry-rep-count-input'
              type='number'
              min='0'
              id='entry-rep-count-input'
              name='entry-rep-count'
              required />
            <br />
            <label htmlFor='entry-weight-count-input'>
              Weight:
            </label>
            <input
              className='entry-weight-count-input'
              type='number'
              min='0'
              id='entry-weight-count-input'
              name='entry-weight-count'
            />
            <br />
            <button
              type='submit'
              className='List-add-button'
            >
              + Add Entry
            </button>
          </form>
          <ul>
            {exerciseLog.map(logEntry => (
              <li
                key={logEntry.id}
              >
                <LogEntry
                  id={logEntry.id}
                  exercise_id={logEntry.exercise_id}
                  weight_count={logEntry.weight_count}
                  set_count={logEntry.set_count}
                  rep_count={logEntry.rep_count}
                  date_logged={logEntry.date_logged}
                  getExerciseLog={this.getExerciseLog}
                />
              </li>
            ))}
          </ul>
        </section>
      </>
    )
  }
}