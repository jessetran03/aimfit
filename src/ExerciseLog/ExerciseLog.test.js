import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import ExerciseLog from './ExerciseLog'
import renderer from 'react-test-renderer'

it('renders ExerciseLog without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
      <ExerciseLog />
    </BrowserRouter>,
    div
  )
  ReactDOM.unmountComponentAtNode(div)
})

it('render ExerciseLog without data', () => {
  const wrapper = renderer.create(
    <BrowserRouter>
      <ExerciseLog />
    </BrowserRouter>
  )
    .toJSON();
    expect(wrapper).toMatchSnapshot()
  })