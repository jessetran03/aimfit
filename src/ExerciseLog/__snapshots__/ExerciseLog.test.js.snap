// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ExerciseLog without data 1`] = `
<section
  className="workouts"
>
  <button
    onClick={[Function]}
  >
    Back
  </button>
  <h2
    className="log-header"
  />
  <form
    className="add-entry-form"
    onSubmit={[Function]}
  >
    <label
      className="entry-set-count-label"
      htmlFor="entry-set-count-input"
    >
      Set x Rep:
    </label>
    <input
      className="entry-set-count-input"
      id="entry-set-count-input"
      min="0"
      name="entry-set-count"
      required={true}
      type="number"
    />
    Â x
    <input
      className="entry-rep-count-input"
      id="entry-rep-count-input"
      min="0"
      name="entry-rep-count"
      required={true}
      type="number"
    />
    <br />
    <label
      htmlFor="entry-weight-count-input"
    >
      Weight:
    </label>
    <input
      className="entry-weight-count-input"
      id="entry-weight-count-input"
      min="0"
      name="entry-weight-count"
      type="number"
    />
    <br />
    <button
      className="List-add-button"
      type="submit"
    >
      + Add Entry
    </button>
  </form>
  <ul />
</section>
`;
